Updating contribution guides for docs. (#325)
magento/magento2-functional-testing-framework: .github/DOCUMENTATION_TEMPLATE.md

# How to contribute to MFTF docs

> We welcome contributions to the MFTF documentation, which is kept within the /docs/ folder in this repository.

Reviewer: ```suggestion
We welcome contributions to the MFTF documentation, which is kept within the `docs/` folder in this repository.
```

This page describes the submitting process and serves as a template for a properly written content.
The contribution workflow is the same as submitting code.

1. Create a branch from the `develop` branch in the [MFTF repository][].
1. Make edits/additions/deletions as needed. Read the [Basic Template][] for tips on how to write with Markdown.

> 1. Submit your PR to the `develop` branch.

Reviewer: ```suggestion
1. Submit your pull request to the `develop` branch.
```

# How to conttribute to MFTF docs

MFTF documentation is kept within the /docs/ folder in this repository.
We welcome contributions to the documentation.
This page describes the process for submitting documentation and serves as a template for a properly written content.

The contribution workflow for the Magento functional testing framework (MFTF) documentation is the same as submitting code.

1. Make a branch from the `develop` branch in the [MFTF repo][].
1. Make edits/additions/deletions as needed.
1. Submit your PR to the `develop` branch.

Once submitted, a member of the documentation team will review and merge it.
We will inform you if it needs any additional processing.

Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

> ## Ordered lists - all heading have a blank line before and after, capitalize the first word only

Reviewer: Move the rule _all heading have a blank line before and after, capitalize the first word only_ to a section content from the heading to be consistent throughout the document.

Reviewer: @dobooth  Any updates on this?

Author: Fixed.

# How to conttribute to MFTF docs

> MFTF documentation is kept within the /docs/ folder in this repo.

Author: Good one.

We welcome contributions to the documentation.
This page describes the process for submitting docs and serves as a template so that docs are written properly.

> The contribution workflow for docs is the same as submitting code.

Author: Thanks

> To contribute to MFTF docs:

Author: Done.

> 1. Make a branch from the [MFTF repo][].

Author: Fixed.

1. Make edits/additions/deletions as needed.
1. Submit your PR to the `develop` branch.

> Once submitted, it will be reviewed by a member of the documentation team.

Author: Fixed.

> If approved it will be tested and merged.

Author: Removed.

> If it needs any work, we will inform you.

Author: Taken.

> Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

Author: Going to leave it in.

> Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

Reviewer: Then you should mention that the documentation is also published on DevDocs website, because it is not clear what Table of Contents do you mean. And that to keep the table of contents consistent, the contributor will need to add the corresponding changes to the devdocs repository.

> Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

Author: Explained and linked to the TOC.

> Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

Reviewer: I don't see the changes.

## H2 heading - blank line before and after, capitalize first word only

> 

Reviewer: ```suggestion

## Ordered lists

```

Author: Added.

> 1. Ordered lists all all numbered 1.

Author: Got it.

> 1. The build process will number them correctly.

Reviewer: ```suggestion
```

Author: Removed.

> 1. Single spafce after the number. Blank line before and after list.

Reviewer: ```suggestion
1. Use single space after the number.
1. Add a blank line before and after list.
```

Author: Got it.

> ### H3

Reviewer: ```suggestion
## Unordered lists
```

Author: Taken.

> - Unordered lists use dashes.

Reviewer: ```suggestion
- Use dashes in unordered lists.
```

Author: Done.

> - One space after the dash.

Reviewer: ```suggestion
- Add one space after the dash.
```

Author: Got it.

> - Blank line before and after list.

Reviewer: ```suggestion
- Add a blank line before and after list.
```

Reviewer: Add an example code block.

Author: Done.

> ## Code Samples

Reviewer: ```suggestion
## Code samples
```
You're breaking your own rule above.

Author: Ugh!

> For formatting and code coloring, wrap code samples in the following format.

Reviewer: ```suggestion
For code highlighting, wrap code samples in the following format:
```

Author: Done.

> Replace the `xml` with the file extension of the code sample type. Use `bash` for command-line text.

Reviewer: ```suggestion
Replace the `xml` with the corresponding language (tupe) of the code sample. Use `bash` for shell commands and `terminal` for terminal output.
```

Reviewer: move this line after the example code block

Author: Done.

```xml
<xmlSample>
   ...
   ...
</xmlSample>
```

## Markdown tables

| Header      | Header |
| ----------- | ----------- |
| Colume 1 text | Column 2 text|
| Column 1 text | Column 2 text|

> Markdown tables work for simple tables. If you need lists or other complex features within a cell, you may have to use a HTML table.

Reviewer: ```suggestion
Markdown formatting works for simple tables. It does not support multiline content within a cell, or split/merged cells within a row or column.
```
We don't want to use complex tables. The content that requires a complex table should be restructured to avoid it.

Author: Thanks.

## Other tips

- Use spaces instead of tabs.

> - One empty line between content. No duplicate empty lines.

Reviewer: ```suggestion
- Do not duplicate blank lines.
```

Author: Got it.

# How to conttribute to MFTF docs

MFTF documentation is kept within the /docs/ folder in this repository.
We welcome contributions to the documentation.
This page describes the process for submitting documentation and serves as a template for a properly written content.

The contribution workflow for the Magento functional testing framework (MFTF) documentation is the same as submitting code.

1. Make a branch from the `develop` branch in the [MFTF repo][].
1. Make edits/additions/deletions as needed.
1. Submit your PR to the `develop` branch.

Once submitted, a member of the documentation team will review and merge it.
We will inform you if it needs any additional processing.

Any changes to the Table of Contents will need to be made in the regular [devdocs repo][].

> ## H2 Heading - blank line before and after, capitalize first word only

Reviewer: The section is empty.
`H2` is obvious for a writer but can be confusing for a contributing developer.
Please consider applying my previous comments.

Author: Ok.

> # How to contribute to MFTF docs

Reviewer: The "documentation" word is used 8 times in this section.
Rephrase to avoid the unneeded repetition. For example:
_We welcome contributions to the MFTF documentation, which is kept within the /docs/ folder in this repository._
_This page describes the submitting process and serves as a template for a properly written content._
_The contribution workflow is the same as submitting code._
_The content in this repository is also a source for the [MFTF guide topics][]._

Author: Done.

MFTF documentation is kept within the /docs/ folder in this repository.
We welcome contributions to the documentation.
This page describes the process for submitting documentation and serves as a template for a properly written content.

The contribution workflow for the Magento functional testing framework (MFTF) documentation is the same as submitting code.

> 1. Create a branch from the `develop` branch in the [MFTF repo][].

Reviewer: Change "repo" to "repository". Avoid abbreviations.

Author: I am slowly learning.

1. Make edits/additions/deletions as needed.
1. Submit your PR to the `develop` branch.

Once submitted, a member of the documentation team will review and merge it.
We will inform you if it needs any additional processing.

The documentation in this repository is used as the source for the [MFTF documentation][].

> Any changes to the Table of Contents will need to be made via a separate PR in the regular [devdocs repo][].

Reviewer: 1. As a developer who is contributing to this repository, how do I know what Table of Contents is this sentence about? There is no Table of Contents here.

2. Change "repo" to "repository". Avoid abbreviations.

Reviewer: As a code contributor, I do not know what is devdocs.

> Any changes to the Table of Contents will need to be made via a separate PR in the regular [Magento Developer documentation repository][].

Reviewer: 1. Replace _via_ with other word.

   See DevDocs style guide:
   > Non-English words and weak phrases
   > Some non-English words and abbreviations are difficult to translate, and some users might be unfamiliar with them.

   > Avoid the following common words and phrases

   > Info
   > Etc. or etcetera
   > e.g.
   > Misc. or miscellaneous
   > Via
   > Make sure
2. It is not clear why _Table of Contents_ is capitalized.
3. Replace _PR_ with _pull request_. Avoid abbreviations. GitHub interface doesn't use this abbreviation, so it can be not clear (GitHub article [About pull requests](https://help.github.com/en/articles/about-pull-requests))

   See DevDocs style guide:

   > Acronyms
   > Use only the most obvious acronyms. For instance, use PHP, HTML, XML for the names of those computer languages, not an unknown acronym. Try to avoid acronyms in your text when possible.

Author: Fixed.
